### ISSUE TITLE
"expected expression, got keyword 'default'" after 0.4.3

### ISSUE THREAD

**Issue by alexandernanberg** [ChristopherBiscardi/gatsby-mdx#317]
**Describe the bug**

When building gatsby I get the following error: 

<details>

```
SyntaxError: Unexpected token default

  - new Function

  - render-page.js:21231 ./node_modules/gatsby-mdx/mdx-renderer.js.module.exports.withMDXComponents
    /Users/alna/dev/strateg/public/render-page.js:21231:46

  - react-dom-server.node.production.min.js:33 c
    [strateg]/[react-dom]/cjs/react-dom-server.node.production.min.js:33:501

  - react-dom-server.node.production.min.js:36 Sa
    [strateg]/[react-dom]/cjs/react-dom-server.node.production.min.js:36:1

  - react-dom-server.node.production.min.js:41 a.render
    [strateg]/[react-dom]/cjs/react-dom-server.node.production.min.js:41:467

  - react-dom-server.node.production.min.js:41 a.read
    [strateg]/[react-dom]/cjs/react-dom-server.node.production.min.js:41:58

  - react-dom-server.node.production.min.js:53 renderToString
    [strateg]/[react-dom]/cjs/react-dom-server.node.production.min.js:53:83

  - render-page.js:825 Module../.cache/static-entry.js.__webpack_exports__.default
    /Users/alna/dev/strateg/public/render-page.js:825:18

  - worker.js:35 Promise
    [strateg]/[gatsby]/dist/utils/worker.js:35:36

  - debuggability.js:313 Promise._execute
    [strateg]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [strateg]/[bluebird]/js/release/promise.js:483:18

  - promise.js:79 new Promise
    [strateg]/[bluebird]/js/release/promise.js:79:10

  - worker.js:31 Promise.map.path
    [strateg]/[gatsby]/dist/utils/worker.js:31:37

  - util.js:16 tryCatcher
    [strateg]/[bluebird]/js/release/util.js:16:23

  - map.js:61 MappingPromiseArray._promiseFulfilled
    [strateg]/[bluebird]/js/release/map.js:61:38

  - promise_array.js:114 MappingPromiseArray.PromiseArray._iterate
    [strateg]/[bluebird]/js/release/promise_array.js:114:31
```

</details>

Also in develop I get the following error:

![image](https://user-images.githubusercontent.com/8997319/54742670-59922780-4bc2-11e9-842f-ffd3d4cc87c7.png)

Edit: Seems to work when I use the 0.5.0 version 

**To Reproduce**

I tried replicating using the [gatsby-mdx minimal repro](https://github.com/ChristopherBiscardi/gatsby-mdx-minimal-repro) but I wasn't able to... 🤔  

Would really appreciate if someone could have a look at my repo anyways (https://github.com/strt/www). Tried stripping away almost everything but still the same error, so I'm a bit lost.

1. Unpin `gatsby-mdx` and install a later version eg. `^0.4.3`
2. Run `yarn dev` and go to `http://localhost:8000/case/save-the-children/`
3. You'll see the `react-error-overlay` with the error above.

**Expected behavior**
Build without errors

**Screenshots**
N/A

**Additional context**
N/A

**Comment by ChristopherBiscardi** [comment]
@alexandernanberg it seems that your site requires a fontawesome token that I don't have so I can't reproduce this.

My best guess is one of the MDX files has content in it that causes this issue. Is there a specific file this happens on?

**Comment by alexandernanberg** [comment]
Ahhh sorry forgot about that! 🤦‍♂️  Created another branch that you *should* be able to run https://github.com/strt/www/tree/gatsby-mdx-repro (if you still have time to).

No, all templates/files with the `MDXRenderer` break, if I comment out the renderer component the page will start to work again. I have already tried without any custom components, no plugins etc. If it will continue to work on `v0.5.0` it's not really a problem though

**Comment by ChristopherBiscardi** [comment]
yeah, so here's the output for one of your pages. Note the `default` without an `export`.

```
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const layoutProps = {};
default class MDXContent extends React.Component {
  constructor(props) {
    super(props);
    this.layout = null;
  }

  render() {
    const _this$props = this.props,
          components = _this$props.components,
          props = _objectWithoutProperties(_this$props, ["components"]);

    return React.createElement(MDXTag, {
      name: "wrapper",
      components: components
    }, React.createElement(MDXTag, {
      name: "column",
      components: components,
      props: {
        "md": 8,
        "bottomGap": "large"
      }
    }, React.createElement(MDXTag, {
      name: "p",
      components: components,
      parentName: "column"
    }, "Familjer\\xE5dgivningen ville n\\xE5 ut med den informationen till nyblivna och blivande f\\xF6r\\xE4ldrar p\\xE5 m\\xF6drav\\xE5rdscentralerna och barnav\\xE5rdscentralerna i Region \\xD6rebro l\\xE4n. De ville skapa igenk\\xE4nning och f\\xF6rbereda m\\xE5lgruppen p\\xE5 de k\\xE4nslor som kan uppst\\xE5. Och ber\\xE4tta att det \\xE4r helt okej att prata med n\\xE5gon \\xE4ven innan det blivit kris och panik."), React.createElement(MDXTag, {
      name: "p",
      components: components,
      parentName: "column"
    }, "Vi intervjuade 20 sm\\xE5barnsf\\xF6r\\xE4ldrar i regionen f\\xF6r att f\\xE5 svar p\\xE5 hur deras relationer p\\xE5verkats. Utifr\\xE5n svaren dramatiserade vi ett antal situationer som tillsammans blev en film som visats p\\xE5 MVC, BVC och i digitala kanaler.")));
  }

}
MDXContent.isMDXComponent = true;
```

**Comment by ChristopherBiscardi** [comment]
I suspect it's related to this replacement missing, which falls through to the next one. Should probably include a replacement for `export default MDXContent`, or we can write a babel plugin that handles the default export replacement with `return` and chops `export` off of the exported consts

https://github.com/ChristopherBiscardi/gatsby-mdx/blob/f16bfb0c0133bcf67564dd956a7c866fd244fef2/packages/gatsby-mdx/utils/gen-mdx.js#L149

**Comment by alexandernanberg** [comment]
Interesting – so just to be clear, this is a bug in `gatsby-mdx` and not in my code? How did you get the output btw?

Thank you for taking the time to debug this 🙌

**Comment by ChristopherBiscardi** [comment]
Yes this is a bug in gatsby-mdx. Should be fixable by adding an additional .replace for `export default MDXContent` -> `return MDXContent`

**Comment by alexandernanberg** [comment]
Fixed in v0.5
